lib/class-site.php

```diff
@@ -159,11 +164,106 @@ public function update_cache() {
 
 		$cache = [];
 		foreach ( $terms as $term ) {
-			$cache[ $term->name ] = [ 'term_id' => $term->term_id, 'slug' => $term->slug ];
+			$cache[ $term->name ] = [
+				'term_id' => $term->term_id,
+				'slug' => $term->slug,
+			];
 		}
 		update_option( 'split_domain_sites', $cache );
 
 		// Unset the cached site term in the Core class.
 		Core::instance()->site_term = null;
 	}
+
+	/**
+	 * UI Overrides for the new/edit term screens for this taxonomy.
+	 */
+	public function ui_override_script() {
+		global $taxnow;
+		if ( $taxnow && $this->name === $taxnow ) :
+			?>
+			<script type="text/javascript">
+			jQuery( function( $ ) {
+				$('.term-name-wrap label,.manage-column.column-name > a > span:first-child').text( <?php echo wp_json_encode( __( 'Domain', 'split-domain' ) ) ?> );
+				$('#tag-name').attr( 'placeholder', 'domain.com' );
+				$('.term-name-wrap p').text( <?php echo wp_json_encode( __( 'The domain, without the protocol (http://) and without a slash at the end. Be sure to include the "www." if the domain will use that.', 'split-domain' ) ) ?> );
+				$('.term-slug-wrap p').text( <?php echo wp_json_encode( __( 'The slug is used for templating. It should be all lowercase and contain only letters, numbers, and hyphens.', 'split-domain' ) ) ?> );
+				$('.term-description-wrap p').text( <?php echo wp_json_encode( __( 'The description is used for internal notes.', 'split-domain' ) ) ?> );
+			});
+			</script>
+			<?php
+		endif;
+	}
+
+	/**
+	 * Validate and normalize a domain string.
+	 *
+	 * @param  string $domain Domain.
+	 * @return string|false   Domain, normalized, if valid; false if invalid.
+	 */
+	protected function validate_domain( $domain ) {
+		$domain = strtolower( $domain );
+		$domain_full = $domain;
+
+		// Prepend protocol for URL validation.
+		if ( ! preg_match( '#^https?://#i', $domain_full ) ) {
+			$domain_full = 'http://' . $domain_full;
+		}
+		$domain = parse_url( $domain_full, PHP_URL_HOST );
+
+		if (
+			empty( $domain )
+			|| ! filter_var( $domain_full, FILTER_VALIDATE_URL )
```

Comment:
`filter_var` with `FILTER_VALIDATE_URL` fails on URLs that contain multibyte characters. If you need to support multibyte characters (international domains etc) you'll need a different validation solution.